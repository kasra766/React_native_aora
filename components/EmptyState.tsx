import { Image, Text, View } from "react-native";
import { images } from "@/constants";
import {CustomButton} from "@/components/CustomButton";
import {router} from "expo-router";

interface IEmptyStateProps {
  title: string;
  subtitle: string;
}

export function EmptyState({ subtitle, title }: IEmptyStateProps) {
  return (
    <View className="justify-center items-center px-4">
      <Image
        source={images.empty}
        className="w-[270px] h-[215px]"
        resizeMode="contain"
      />
      <Text className="text-xl text-center mt-2 font-psemibold text-white ">
        {title}
      </Text>
      <Text className="font-pmedium text-sm text-gray-100">{subtitle}</Text>
        <CustomButton title="Create video" handlePress={()=>router.push('/create')}
        containerStyles='w-full my-5'
        />
    </View>
  );
}
