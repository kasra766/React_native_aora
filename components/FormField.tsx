import React, { useState } from "react";
import { Text, TextInput, TouchableOpacity, View } from "react-native";
import { TabBarIcon } from "@/components/navigation/TabBarIcon";

interface IFormFieldProps
  extends React.ComponentPropsWithoutRef<typeof TextInput> {
  label: string;
  otherStyles?: string;
}
export function FormField({
  label,
  otherStyles,
  ...inputProps
}: IFormFieldProps) {
  const [showPassword, setShowPassword] = useState(false);
  const togglePassword = () => setShowPassword(!showPassword);
  return (
    <View className={`space-y-2 ${otherStyles}`}>
      <Text className="text-base text-gray-100 font-pmedium">{label}</Text>
      <View className="w-full h-16 px-4 bg-black-100 rounded-2xl border-2 border-black-200 focus:border-secondary flex flex-row items-center">
        <TextInput
          className="flex-1 text-white font-psemibold text-base"
          secureTextEntry={label === "Password" && !showPassword}
          {...inputProps}
        />
        {label === "Password" && (
          <TouchableOpacity onPress={togglePassword}>
            <TabBarIcon name={showPassword ? "eye-off" : "eye"} color='#fff'/>
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
}
