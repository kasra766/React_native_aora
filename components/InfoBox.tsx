import { Text, View } from "react-native";

interface IProps {
  title: string | number;
  containerStyles?: string;
  subtitle?: string;
  titleStyles?: string;
}
export function InfoBox({
  titleStyles,
  title,
  containerStyles,
  subtitle,
}: IProps) {
  return (
    <View className={containerStyles}>
      <Text className={`text-white text-center font-psemibold ${titleStyles}`}>
        {title}
      </Text>
      <Text className="text-sm text-gray-100 text-center font-pregular">
        {subtitle}
      </Text>
    </View>
  );
}
