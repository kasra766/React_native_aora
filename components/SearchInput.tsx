import React, { useState } from "react";
import { Alert, Text, TextInput, TouchableOpacity, View } from "react-native";
import { TabBarIcon } from "@/components/navigation/TabBarIcon";
import { router, usePathname } from "expo-router";

interface ISearchInputProps
  extends React.ComponentPropsWithoutRef<typeof TextInput> {
  initialValue?: string;
}
export function SearchInput({
  initialValue,
  ...inputProps
}: ISearchInputProps) {
  const pathName = usePathname();
  const [query, setQuery] = useState(initialValue || "");

  function handleSubmitQuery() {
    if (!query.trim()) {
      return Alert.alert(
        "Missing query",
        "Please input something to search results across database",
      );
    }
    if (pathName.startsWith("/search")) return router.setParams({ query });
    router.push(`/search/${query}`);
  }
  return (
    <View className="flex flex-row items-center space-x-4 w-full  px-4 py-2 bg-black-100 rounded-2xl border-2 border-black-200 focus:border-secondary">
      <TextInput
        className="text-base  text-white flex-1 font-pregular"
        placeholder="Search a video topic"
        placeholderTextColor="#CDCDE0"
        {...inputProps}
        onChangeText={setQuery}
        testID='searchInputField'
      />

      <TouchableOpacity onPress={handleSubmitQuery}>
        <TabBarIcon name={"search"} color="#fff" className="size-5" />
      </TouchableOpacity>
    </View>
  );
}
