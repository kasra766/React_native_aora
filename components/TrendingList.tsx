import {
  FlatList,
  Image,
  ImageBackground,
  Text,
  TouchableOpacity,
  ViewToken,
} from "react-native";
import * as Animatable from "react-native-animatable";
import { useState } from "react";
import { TabBarIcon } from "@/components/navigation/TabBarIcon";
interface ITrendingListProps {
  posts: any[];
}
interface ITrendingItemProps {
  item: ITrendingListProps["posts"][number];
  activeItemId: number | string;
}

const zoomIn = {
  0: {
    scale: 0.9,
  },
  1: {
    scale: 1.1,
  },
};
const zoomOut = {
  0: {
    scale: 1,
  },
  1: {
    scale: 0.9,
  },
};

function TrendingItem({ item, activeItemId }: ITrendingItemProps) {
  const [play, setPlay] = useState(false);
  return (
    <Animatable.View
      className="mr-5"
      animataion={activeItemId === item.$id ? zoomIn : zoomOut}
      duration={500}
    >
      {play ? (
        <Text>Playing</Text>
      ) : (
        <TouchableOpacity
          activeOpacity={0.7}
          onPress={() => setPlay(!play)}
          className="relative justify-center items-center "
        >
          <ImageBackground
            source={{ uri: item.thumbnail }}
            className="w-52 h-72 rounded-[35px] my-5 overflow-hidden shadow-lg shadow-black/40"
            resizeMode="cover"
          />
          <TabBarIcon
            name="play-circle-outline"
            className="absolute w-12 h-12"
            color="#fff"
          />
        </TouchableOpacity>
      )}
    </Animatable.View>
  );
}
export function TrendingList({ posts }: ITrendingListProps) {
  const [activeItem, setActiveItem] = useState(posts[0]);
  const onViewableItemChanged = (info: {
    viewableItems: ViewToken<any>[];
    changed: ViewToken<any>[];
  }) => {
    if (info.viewableItems.length > 0) {
      setActiveItem(info.viewableItems[0].key);
    }
  };
  return (
    <FlatList
      data={posts}
      keyExtractor={(item) => item.$id.toString()}
      renderItem={({ item }) => (
        <TrendingItem item={item} activeItemId={activeItem} />
      )}
      onViewableItemsChanged={onViewableItemChanged}
      viewabilityConfig={{
        itemVisiblePercentThreshold: 70,
      }}
      contentOffset={{ x: 170, y: 0 }}
      horizontal
    />
  );
}
