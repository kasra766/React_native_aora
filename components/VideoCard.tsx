import { Image, Text, TouchableOpacity, View } from "react-native";
import { TabBarIcon } from "@/components/navigation/TabBarIcon";
import React, { useState } from "react";
import { ResizeMode, Video } from "expo-av";

interface IVideoProps {
  title: string;
  creator: string;
  avatar: string;
  thumbnail: string;
  video: string;
}
export function VideoCard({
  title,
  creator,
  avatar,
  thumbnail,
  video,
}: IVideoProps) {
  const [play, setPlay] = useState(false);

  return (
    <View className="flex-col items-center px-4 mb-14">
      <View className="flex flex-row gap-3 items-start">
        <View className="flex justify-center items-center flex-row flex-1">
          <View className="w-[46px] h-[46px] rounded-lg border border-secondary flex justify-center items-center p-0.5">
            <Image
              source={{ uri: avatar }}
              className="w-full h-full rounded-lg"
              resizeMode="cover"
            />
          </View>
          <View className="flex justify-center flex-1 ml-3 gap-y-1">
            <Text
              className="font-psemibold text-sm text-white"
              numberOfLines={1}
            >
              {title}
            </Text>
            <Text
              className="text-xs text-gray-100 font-pregular"
              numberOfLines={1}
            >
              {creator}
            </Text>
          </View>
        </View>
        <View className="pt-2">
          <TabBarIcon name="ellipsis-vertical" color="#fff" />
        </View>
      </View>

      {play ? (
        <Video
          source={{ uri: video }}
          className="w-full h-60 rounded-xl mt-3 bg-white/10"
          resizeMode={ResizeMode.CONTAIN}
          useNativeControls
          shouldPlay
          onPlaybackStatusUpdate={(status) => {
            //@ts-ignore
            if (status.didJustFinish) {
              setPlay(false);
            }
          }}
        />
      ) : (
        <TouchableOpacity
          activeOpacity={0.7}
          onPress={() => setPlay(!play)}
          className="w-full h-60 rounded-xl mt-3 relative justify-center items-center"
        >
          <Image
            source={{ uri: thumbnail }}
            resizeMode="cover"
            className="w-full h-full rounded-xl mt-3"
          />
          <TabBarIcon
            name="play-circle-outline"
            className="absolute w-12 h-12"
            color="#fff"
          />
        </TouchableOpacity>
      )}
    </View>
  );
}
