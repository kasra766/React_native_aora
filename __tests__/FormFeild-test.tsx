import { fireEvent, render, screen } from "@testing-library/react-native";
import { FormField } from "@/components/FormField";

describe("formfield tests", () => {
  test("renders label", () => {
    render(<FormField label="test label" />);
    const label = screen.getByTestId("label");
    expect(label).toHaveTextContent("test label");
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test("renders input", () => {
    const onChange = jest.fn();
    render(<FormField label="test label" onChangeText={onChange} />);
    const input = screen.getByTestId("input");
    fireEvent.changeText(input, "new");
    expect(onChange).toHaveBeenCalledWith("new");
  });
});
