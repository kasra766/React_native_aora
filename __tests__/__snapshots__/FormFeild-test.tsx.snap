// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFeild Test renders correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#CDCDE0",
        },
        {
          "fontFamily": "Poppins-Medium",
        },
      ]
    }
    testID="label"
  >
    test label
  </Text>
  <View
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      [
        {
          "width": "100%",
        },
        {
          "height": 64,
        },
        {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        {
          "backgroundColor": "#1E1E2D",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#232533",
          "borderLeftColor": "#232533",
          "borderRightColor": "#232533",
          "borderTopColor": "#232533",
        },
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        [
          {
            "marginTop": 8,
          },
        ],
      ]
    }
  >
    <TextInput
      secureTextEntry={false}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "color": "#fff",
          },
          {
            "fontFamily": "Poppins-SemiBold",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`formfield tests renders label 1`] = `
<View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#CDCDE0",
        },
        {
          "fontFamily": "Poppins-Medium",
        },
      ]
    }
    testID="label"
  >
    test label
  </Text>
  <View
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      [
        {
          "width": "100%",
        },
        {
          "height": 64,
        },
        {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        {
          "backgroundColor": "#1E1E2D",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#232533",
          "borderLeftColor": "#232533",
          "borderRightColor": "#232533",
          "borderTopColor": "#232533",
        },
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        [
          {
            "marginTop": 8,
          },
        ],
      ]
    }
  >
    <TextInput
      secureTextEntry={false}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "color": "#fff",
          },
          {
            "fontFamily": "Poppins-SemiBold",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
      testID="input"
    />
  </View>
</View>
`;
